runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,7 +41,7 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       if !AsyncOperationStatus.is_terminal_status(polling_state.status)
         task = Concurrent::TimerTask.new do
           begin
-            if !polling_state.azure_async_operation_header_link.nil?
+            if !polling_state.azure_async_operation_header_link.nil? && custom_deserialization_block.nil?
```

Comment:
@veronicagg 
1. If we send a custom deserialization block, 
2. the response does have a async operation header link 
3. No location header link 
4. And Not a Put Method

Yes. I agree that would be an issue. But is there a scenario where only one of the links is provided and the other one is not present?